tarsier-service/service/src/server.py

```diff
@@ -0,0 +1,63 @@
+from fastapi import FastAPI, Depends
+from fastapi.middleware.cors import CORSMiddleware
+from browser import url_to_page_text
+import llm
+import db
+from propelauth_fastapi import User
+from auth import validate_api_key, auth
+from models import PageTextData, ExtractData, ExtractResponseData
+import stripe
+
+CLIENT_DOMAIN = "http://localhost:3000"
+
+app = FastAPI()
+origins = [CLIENT_DOMAIN]
+app.add_middleware(CORSMiddleware, allow_origins=origins, allow_credentials=True, allow_methods=["*"], allow_headers=["*"])
+
+ 
+@app.post("/page-text")
+async def get_page_text(data: PageTextData, user: User = Depends(validate_api_key)) -> str: 
+    text = await url_to_page_text(data.url)
+    db.create_page_text_job(user["user_id"], data.url, text)
+    return text
+
+
+@app.post("/extract")
+async def extract_data(data: ExtractData, user: User = Depends(validate_api_key)) -> ExtractResponseData:
+    page_text = await url_to_page_text(data.url)
+    extracted_data = await llm.structured_data_from_page_text(page_text, data.outputSchema)
+    print(extracted_data)
+
+    response = { 'data': extracted_data }
+    if (data.options and data.options.return_page_text):
+        response['page_text'] = page_text
+
+    db.create_extraction_job(user["user_id"], data.url, page_text, data.outputSchema, extracted_data)
```

Comment:
nit: Fine to leave this for now but in practice, we want all dependencies to be injected to the function as opposed to relying on global `llm` or `db` objects